commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/localization/LocalizationServiceImp.java

```
@@ -0,0 +1,124 @@
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Created by Racha on 29/01/2016.
+ */
+public class LocalizationServiceImp implements LocalizationService {
+
+
+    private static final Log LOG = ExoLogger.getLogger(LocalizationServiceImp.class);
+    private String lang = null;
+
+    @Override
+   public String getLanguage(ExoContainer container,Identity currentUser,  HttpServletRequest request ){
+        try {
+            LocaleConfigService localeConfigService = (LocaleConfigService) container.getComponentInstanceOfType(LocaleConfigService.class);
+            LocalePolicy localePolicy = (LocalePolicy) container.getComponentInstanceOfType(LocalePolicy.class);
+            LocaleContextInfo localeCtx = new LocaleContextInfo();
+            Locale portalLocale = Locale.getDefault();
+            Set<Locale> supportedLocales = new HashSet();
+
+            for (LocaleConfig lc : localeConfigService.getLocalConfigs()) {
+                supportedLocales.add(lc.getLocale());
+            }
+
+            localeCtx.setSupportedLocales(supportedLocales);
+            localeCtx.setBrowserLocales(Collections.list(request.getLocales()));
+            localeCtx.setCookieLocales(LocalizationLifecycle.getCookieLocales(request));
+            localeCtx.setSessionLocale(LocalizationLifecycle.getSessionLocale(request));
+            localeCtx.setUserProfileLocale(getUserProfileLocale(container, currentUser.getUserId()));
+            localeCtx.setRemoteUser(currentUser.getUserId());
+
+            localeCtx.setPortalLocale(checkPortalLocaleSupported(portalLocale, supportedLocales));
+            Locale locale = localePolicy.determineLocale(localeCtx);
+            lang = locale.getLanguage();
+
+        } catch (Exception e) {
+            LOG.error("Cannot get the current language of the user " + currentUser.getUserId());
```

Do not swallow exceptions. It should be : LOG.error("Cannot get the current language of the user " + currentUser.getUserId(), e);